azurerm/internal/services/iotcentral/reosurce_arm_iotcentral_aplication.go

```
@@ -0,0 +1,293 @@
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.IotCentralName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"location": azure.SchemaLocation(),
+
+			"sub_domain": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.IotCentralSubDomain,
+			},
+
+			"display_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.IotCentralDisplayName,
+			},
+
+			"sku": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(iotcentral.S1), //Currently only support S1. The F1 Sku is no longer supported
+				}, true),
+			},
+			"template": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"tags": tags.Schema(),
+		},
+	}
+}
+
+func resourceArmIotCentralAppCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).IoTCentral.AppsClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroup, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing IoT Central Application  %q (Resource Group %q): %+v", name, resourceGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_iotcentral_application", *existing.ID)
+		}
+	}
+
+	req, err := client.CheckNameAvailability(ctx, iotcentral.OperationInputs{
```

This call is not necessary, since if its in avaliable, the `CreateOrUpdate` will fail anyway.

---

I think name check is need here. Request will fail fast if name not available until send the create request.